### PR TITLE
INSP: don't search initial items for auto import in transitive dependencies

### PR DESCRIPTION
**Pull request by Undin**
* Introduce cargo project search scope
* Use this scope to exclude transitive dependencies' items from auto import candidates

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt [hunk 1]
```diff
@@ -27,6 +31,9 @@ interface CargoWorkspace {
 
     val packages: Collection<Package>
     fun findPackage(name: String): Package? = packages.find { it.name == name }
+    fun findPackageForFile(file: VirtualFile): Package?
+    /** for internal usage only */
+    fun resetPackageIndex()
```
Comment (Undin) [inline src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt]: I don't like this method in public API but I can't come up with a better way to allow `CargoProjectsServiceImpl` call index reset

**Comment by Undin** [comment]
@vlad20012 Done

File: src/main/kotlin/org/rust/cargo/project/model/impl/CargoPackageIndex.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                }
+            })
+        }
+        indexDisposable = disposable
+    }
+
+    fun findPackageForFile(file: VirtualFile): Package? {
+        checkReadAccessAllowed()
+        val cargoProject = service.findProjectForFile(file) ?: return null
+        return indices[cargoProject]?.getInfoForFile(file)?.orElse(null)
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/cargo/project/model/impl/CargoPackageIndex.kt]: Can we omit `?.orElse(null)` here?

Comment (Undin) [reply src/main/kotlin/org/rust/cargo/project/model/impl/CargoPackageIndex.kt]: But we need to extract `Package` object from `Optional` and looks like it's the simplest way to do it

**Pull request merged**